lib/commons/builder/config.rb

```diff
@@ -6,11 +6,14 @@
 class Config
   attr_reader :values
 
-  def initialize(file)
-    @values = JSON.parse(
-      Pathname.new(file).read,
-      symbolize_names: true
-    )
+  def initialize(values)
+    raise 'values should be a Hash' unless values.is_a? Hash
```

Comment:
I know that it seems like good defensive programming to add this explicit type check, but it's not very Ruby-ish, I think - you tend to trust the caller, and this allows the caller to exploit duck-typing to pass in something other than a Hash that still responds to `[]`. (If you really want to check, then I think `responds_to?` is preferred to `is_a?`, but I don't think either is really needed here. I'm sure other opinions are available, as they say :smiley:)

Reply:
As above (https://github.com/everypolitician/commons-builder/pull/58#discussion_r193034138) :-)